### ISSUE TITLE
Error generate k8s when using symlinks 

### ISSUE THREAD

**Issue by allamand** [operator-framework/operator-sdk#1411]
## Bug Report

I've got an error while generating k8s code when the code is symlink in the target directory

i'm using version 0.7.0

**What did you do?**

Without symlink it works :
```
docker run -ti --rm --privileged -v /Users/go/src/path/to/project:/go/src/path/to/project -w /go/src/path/to/project \
        build-operator:sdk-0-7-0 bash
Starting Docker daemon in DinD mode in background...
Docker daemon up and running:
  Server Version: 18.06.1-ce
  Storage Driver: vfs
root@a9c73ff036dc:/go/src/path/to/project# operator-sdk generate k8s
INFO[0019] Running deepcopy code-generation for Custom Resource group versions: [db:[v1alpha1], ]
INFO[0030] Code-generation complete.
```

With symlinks it broke : 

```
docker run -ti --rm --privileged -v /Users/go/src/path/to/project:/source -w /source \
  --env https_proxy= --env http_proxy= \
        build-operator:sdk-0-7-0 bash
Starting Docker daemon in DinD mode in background...
Docker daemon up and running:
  Server Version: 18.06.1-ce
  Storage Driver: vfs
root@0fd5e58b1e1f:/source# mkdir -p $GOPATH/src/path/to/project
root@0fd5e58b1e1f:/source# ln -s $(pwd) $GOPATH/src/path/to/project
root@0fd5e58b1e1f:/source# cd $GOPATH/src/path/to/project
root@0fd5e58b1e1f:/go/src/path/to/project# operator-sdk generate k8s
Error: exit status 1
Usage:
  operator-sdk generate k8s [flags]

Flags:
      --header-file string   Path to file containing headers for generated files.
  -h, --help                 help for k8s
```

in my pipeline I can't force the working_directory so I go through the symlink, but it won't work anymore.

thanks for help

**Comment by joelanford** [comment]
@allamand Has this worked in earlier versions of `operator-sdk`?

We've seen issues with symlinks come up relatively often, and the answer is typically to flip the symlink and make sure that the symlink points into $GOPATH, rather than creating a symlink in $GOPATH pointed somewhere outside $GOPATH. 

Is doing something like that possible in your pipeline?

**Comment by allamand** [comment]
Hi @joelanford, yes it was working in the previous version. For now, I've worked out it by copying files in the GOPATH instead of making symlinks. It take longer to make copy in the pipeline, but it works :)

**Comment by joelanford** [comment]
@allamand I think the issue is a result of changes we've been making to the way we build code generation binaries. As of this morning, we merged #1470 which vendors the codegen libraries into the SDK binary, so we no longer need to build binaries in user projects.

With this change, you _may_ be able to revert back to using symlinks. I wouldn't be surprised if go modules, which we now support since `v0.8.0`, also improves symlink usage.

Since it sounds like you have a workaround and we may have fixed this in #1470, I'll go ahead and close this.If you try out #1470 and go modules and the symlink doesn't work, definitely re-open and we can try to help out.
